### PR TITLE
Prepare for deprecation of ViewAllocateWithoutInitializing

### PR DESCRIPTION
**Pull request by dalg24**
Per comment in the code, `ViewAllocateWithoutInitializing` was there only for backward compatibility.  Moving away from it allows us to get rid of two `View` constructor overloads*.
~~Admittedly the changes I propose would break code that relied on `Kokkos::ViewAllocateWithoutInitializing` being an actual type but I think it is ok.~~

I considered adding an implicit conversion from `ViewAllocateWithoutInitializing` to `Impl::ViewCtorProp<std::string,Impl::WithoutInitializing_t>` because that solution would be safer for code downstream.  In fact, it was originally my plan but it turned out to be harder than I thought.  I managed to implement it with the help from @nliber but it forced me to change all the overloads that took `Impl::ViewCtorProp<P...>` and I ended dropping the idea because the complexity added was not worth it.

You will notice I haven't actually yet deprecated the use of `ViewAllocateWithoutInitializing`.   My intent is to make sure we remove it from all examples and tutorials and perform the deprecation later.  Hopefully by then we will be able to use the deprecated attribute from C++14.

**TLDR** Make `ViewAllocateWithoutInitializing` a ~~function that generates a `Impl::ViewCtorProp` rather a concrete type~~ **an alias** in preparation for deprecation/removal.

*Don't worry we still have 14 besides besides the special members!

### CODE REVIEW

**Comment by dalg24** [comment]
I had a peek at Trilinos and this PR would break
```
packages/muelu/test/scaling/MMKernelDriver.cpp:282:    using no_init_view=Kokkos::ViewAllocateWithoutInitializing;
packages/muelu/test/scaling/MMKernelDriver.cpp:442:    using no_init_view=Kokkos::ViewAllocateWithoutInitializing;
packages/muelu/test/scaling/MMKernelDriver.cpp:689:        using no_init_view=Kokkos::ViewAllocateWithoutInitializing;
packages/muelu/test/scaling/MMKernelDriver.cpp:842:    using no_init_view=Kokkos::ViewAllocateWithoutInitializing;
packages/shylu/shylu_node/tacho/src/impl/Tacho_Util.hpp:531:  using do_not_initialize_tag = Kokkos::ViewAllocateWithoutInitializing;
```
one possible fix would be `Kokkos::ViewAllocateWithoutInitializing => decltype(Kokkos::view_alloc(Kokkos::WithoutInitializing, "whatever"))`

**Comment by dalg24** [comment]
@crtrott encouraged me to come up with a solution that does not break code.  So here it is.

Make sure to drop 730adc2 before we merge.  It is only there to demonstrate that the solution I propose works.

**Comment by dalg24** [comment]
Build before dropping the commit log is [here](https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2494/pipeline)

I added a unit test anyway.

**Comment by nliber** [comment]
Thoroughly looked through the changes.  Looks good to me.
